src/Npgsql/NpgsqlCommand.cs

```diff
@@ -195,6 +195,26 @@ public override string CommandText
         }
     }
 
+    string TracingCommandText => IsWrappedByBatch ? GetBatchFullCommandText() : CommandText;
+
+    string GetBatchFullCommandText()
+    {
+        Debug.Assert(IsWrappedByBatch);
+        if (InternalBatchCommands.Count == 1)
+            return InternalBatchCommands[0].CommandText;
+        // TODO: Potentially cache on connector/command?
```

Comment:
Yeah, we also allocate a StringBuilder in SqlQueryParser, we could have a common one on the connector indeed...